esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -689,6 +689,17 @@ extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSoc
     }
 }
 
+extern TPWRAPPER_API void queryQueriesOnlyRoxieTargetNames(StringArray& names)
```

Comment:
I would probably name this "get" rather than "query" since you are building a list.

But I wonder whether you really want "QueriesOnlyRoxie".  Current cloud makes it seem like there are only 2 types but remember in bare-metal there are three types, and it may become like that again in cloud some day.

enum RoxieTargetType
{
    RTTUnknown = 0,
    RTTPublished = 1,
    RTTQueued = 2,
    RTTBoth = 3
};
I think RTTPublished means the same as "QueriesOnly" and "RTTQueued" is like "roxie-workunit.

But RTTBoth can also have queries on it.

What it boils down to for cloud is just the function name, but for bare-metal we should probably support RTTBoth and RTTPublished.

This might be too long but the idea would be a function name like "getRoxieTargetsSupportingPublishedQueries()".